lib/jsdom/living/nodes/Node-impl.js

```diff
@@ -181,19 +181,19 @@ class NodeImpl extends EventTargetImpl {
       newChildImpl._ownerDocument = this._ownerDocument;
     }
 
-    // TODO - if (!newChild) then?
-    if (this.nodeName !== "#document" && newChildImpl._ownerDocument !== this._ownerDocument) {
-      throw new DOMException(DOMException.WRONG_DOCUMENT_ERR);
-    }
-
     if (newChildImpl.nodeType && newChildImpl.nodeType === NODE_TYPE.ATTRIBUTE_NODE) {
       throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR);
     }
 
-    // search for parents matching the newChild
-    for (const ancestor of domSymbolTree.ancestorsIterator(this)) {
-      if (ancestor === newChildImpl) {
-        throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR);
+    if (this._ownerDocument !== newChildImpl._ownerDocument) {
+      // adopt the node when it's not in this document
+      this._ownerDocument.adoptNode(newChildImpl);
+    } else {
+      // search for parents matching the newChild
+      for (const ancestor of domSymbolTree.ancestorsIterator(this)) {
```

Comment:
If the node was in another document and we are adopting it, it cannot be an ancestor of `newChildImpl`. I suppose it could be left as it was, but for that case the search will always be fruitless.